dump_offload_handler.cpp

```diff
@@ -0,0 +1,74 @@
+#include "dump_offload_handler.hpp"
+
+#include "dump_dbus_util.hpp"
+#include "dump_send_pldm_cmd.hpp"
+#include "dump_utility.hpp"
+
+#include <fmt/format.h>
+
+#include <phosphor-logging/log.hpp>
+
+namespace openpower::dump
+{
+using ::openpower::dump::utility::DumpType;
+using ::openpower::dump::utility::ManagedObjectType;
+using ::phosphor::logging::level;
+using ::phosphor::logging::log;
+
+DumpOffloadHandler::DumpOffloadHandler(sdbusplus::bus::bus& bus,
+                                       const std::string& entryIntf,
+                                       DumpType dumpType) :
+    _sdbus(bus),
+    _entryIntf(entryIntf), _dumpType(dumpType),
+    _dumpWatch(bus, entryIntf, dumpType)
+{
+}
+
+void DumpOffloadHandler::offload()
+{
+    try
+    {
+        log<level::INFO>(
+            fmt::format(
+                "DumpOffloadHandler::offload entryType ({}) dumpType ({})",
+                _entryIntf, _dumpType)
+                .c_str());
+        ManagedObjectType objects = openpower::dump::getDumpEntries(_sdbus);
+        for (auto& object : objects)
+        {
+            std::string strObjPath = object.first;
```

Comment:
You could avoid this variable, we can use `object.first` right?

Reply:
objject.first is of type object_path which I need to conver to std::string to put it in logs.

Reply:
FYA, The object_path has the members called `str` to get the string type value
https://github.com/openbmc/sdbusplus/blob/master/include/sdbusplus/message/native_types.hpp#L87 - not expecting to fix it now, just sharing.